app/src/main/java/com/iic/shopingo/ui/HomeActivity.java

```diff
@@ -21,14 +34,61 @@
   @InjectView(R.id.home_action_card_make_request)
   ActionCardView makeRequestCard;
 
+  @InjectView(R.id.home_content_container)
+  View contentContainer;
+
+  @InjectView(R.id.home_spinner_container)
+  View spinnerContainer;
+
   @Override
   protected void onCreate(Bundle savedInstanceState) {
     super.onCreate(savedInstanceState);
-    if (!SharedUserConnector.getInstance().isUserSignedIn()) {
+    init();
+  }
+
+  @Override
+  protected void onStart() {
+    super.onStart();
+    if (CurrentUser.getInstance().state == CurrentUser.State.LOGGED_OUT) {
       navigateToOnboarding();
     } else {
-      // TODO: check user state and go to the correct screen directly.
-      init();
+      showSpinner();
+
+      new GetCurrentState(CurrentUser.getToken()).executeAsync().continueWith(new Continuation<CurrentStateApiResult, Void>() {
+        @Override
+        public Void then(Task<CurrentStateApiResult> task) throws Exception {
+          if (!task.isFaulted() && !task.isCancelled()) {
+            switch (task.getResult().userState) {
+              case LOGGED_OUT: {
+                Log.e("HomeActivity", "Server should not report user as logged out");
+                finish();
+              } break;
```

Comment:
You don't need the curly braces in each `case` clause, you can remove it.
`break` is used to separate between the case statements.